File: algorithms/2_medium/B/00991_broken_calculator/solution.c
int brokenCalc(const int start_value, const int target) {
    int cnt = 0;
    int mul = 1;

    while (start_value * mul < target) {
        mul <<= 1;
        cnt += 1;
    }

    int diff = start_value * mul - target;

    while (diff != 0) {
        cnt += diff / mul;
        diff = diff % mul;
        mul >>= 1;
    }

    return cnt;
}

File: algorithms/2_medium/B/00991_broken_calculator/solution.rs
impl Solution {
    pub fn broken_calc(start_value: i32, target: i32) -> i32 {
        let mut cnt = 0_i32;
        let mut mul = 1_i32;

        while start_value * mul < target {
            mul <<= 1;
            cnt += 1;
        }

        let mut diff = start_value * mul - target;

        while diff != 0 {
            cnt += diff / mul;
            diff = diff % mul;
            mul >>= 1;
        }

        cnt
    }
}
